src/idle-timer.c

```diff
@@ -281,7 +281,7 @@ notice_events_timer (gpointer closure)
   XSetErrorHandler (old_handler);
 
   /* we pop once, and we don't pop again */
-  return 0;
+  return G_SOURCE_REMOVE;
```

Comment:
Could you please change the return type of this function to `gboolean`? Then we are fully in line with `GSourceFunc` signature (not that it makes any difference in effect, but it avoids confusion).

Reply:
I guess `check_pointer_timer` and `idle_timeout_main_loop` are candidates for the same treatment.

Reply:
Sounds good. I will wait for the reformatting to land first.

Reply:
You should be good to go :-)